BinaryClock/Display.cpp

```diff
@@ -46,17 +46,53 @@ void Display::print_static_text(const String &text)
     delay(1000);
 }
 
+ // void Display::print_time_binary(const Time &val)
+ // {
+ //     MATRIX.clear();
+ //
+ //     String digits = val.to_binary_string();
+ //     for (int i = 0; i < digits.length(); i++) {
+ //         if (digits.charAt(i) == '1') {
+ //             unsigned height = MATRIX.height();
+ //             MATRIX.drawFastVLine(i, 0, height, LED_ON);
+ //         }
+ //     }
+ //
+ //     MATRIX.writeDisplay();
+ // }
+
 void Display::print_time_binary(const Time &val)
 {
-    MATRIX.clear();
+   MATRIX.clear();
 
-    String digits = val.to_binary_string();
-    for (int i = 0; i < digits.length(); i++) {
-        if (digits.charAt(i) == '1') {
-            unsigned height = MATRIX.height();
-            MATRIX.drawFastVLine(i, 0, height, LED_ON);
-        }
-    }
+   String digits = val.to_binary_string();
+   for (int i = 0; i < 5; i++)
+   {
+       if (digits.charAt(i) == '1'){
+       unsigned height = MATRIX.height();
+       MATRIX.drawFastVLine(i, 0, height, LED_ON);
```

Comment:
The first for loop only does the first 5 digits, the hour.

Then the colons,

Then the last 6 lines are printed in the second for loop which fill the remaining 6 rows.

Unless I understood incorrectly what right aligned means